src/app/services/push.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { ApiResponse } from '../models/api-response';
+@Injectable({
+	providedIn: 'root'
+})
+/*
+* The PushService is called on SEARCH_COMPLETE_SUCCESS
+* to push the result data back to server in the same
+* search result object format obtained from server.
+**/
+export class PushService {
+
+	constructor( private http: HttpClient ) { }
+
+	public postData(data: ApiResponse) {
```

Comment:
Shouldn't this method use some callback or return a promise? How will other parts of the code interact with the completion state of this method?

Reply:
I guess we do not need a call back or a return promise since we are not currently using the returned response apart from logging it on console.

But surely, we can return the response. It can be used to test the method/push service, and I would be adding it while writing test case for it.